src/processTargets/modifiers/scopeHandlers/TokenScopeHandler.ts

```diff
@@ -43,9 +43,9 @@ export default class TokenScopeHandler extends NestedScopeHandler {
     } = scopeA;
     const { identifierMatcher } = getMatcher(document.languageId);
 
-    return identifierMatcher.test(document.getText(scopeA.domain))
+    return testRegex(identifierMatcher, document.getText(scopeA.domain))
```

Comment:
```suggestion
    // NB: Don't directly use `test` here because global regexes are stateful
    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#finding_successive_matches
    return testRegex(identifierMatcher, document.getText(scopeA.domain))
```